crates/belly_widgets/src/input/text.rs

```diff
@@ -271,12 +310,26 @@ fn process_keyboard_input(
                     chars.remove(index);
                     input.value = chars.iter().collect();
                 }
+            } else if ch.char == CHAR_DELETE {
+                if !selected.is_empty() {
+                    chars.drain(selected.range());
+                    index = selected.min;
+                    selected.stop();
+                    input.value = chars.iter().collect();
+                } else {
+                    if chars.len() > index {
+                        chars.remove(index);
+                        input.value = chars.iter().collect();
+                    }
+                }
             } else {
                 if !selected.is_empty() {
                     chars.drain(selected.range());
                     index = selected.min;
                     selected.stop();
                 }
+                // this will try to insert anything, even if it doesnt have a textual representaion
+                // (escape for exampe will be inserted as a space) some constraint would be useful
```

Comment:
This is a good point. I'd appreciate if you apply a fix for this part, something like replacing `} else {` with 
```rust
// LAST_CONTROL_CHARACTER is '\u{1f}'
} else if ch.char > LAST_CONTROL_CHARACTER {
```